magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Misc.scala

```diff
@@ -433,27 +434,53 @@ package misc {
 
     implicit val queryRegionFormat: RootJsonFormat[QueryRegion] = jsonFormat2(QueryRegion.apply)
 
-    class RegionFormat(bbFormat: JsonFormat[BoundingBox]) extends JsonFormat[Region] {
-      override def write(region: Region): JsValue = JsObject(Map(
-        "regionId" -> region.queryRegion.regionId.toJson,
-        "regionType" -> region.queryRegion.regionType.toJson,
-        "regionName" -> region.regionName.toJson,
-        "regionShortName" -> region.regionShortName.toJson,
-        "boundingBox" -> region.boundingBox.map(_.toJson(bbFormat)).toJson).filter(x => !x._2.equals(JsNull)))
+    class RegionFormat(bbFormat: JsonFormat[BoundingBox])
+        extends JsonFormat[Region] {
+      override def write(region: Region): JsValue =
+        JsObject(
+          Map(
+            "regionId" -> region.queryRegion.regionId.toJson,
+            "regionType" -> region.queryRegion.regionType.toJson,
+            "regionName" -> region.regionName.toJson,
+            "regionShortName" -> region.regionShortName.toJson,
+            "boundingBox" -> region.boundingBox.map(_.toJson(bbFormat)).toJson,
+            "steId" -> region.steId.toJson,
+            "sa4Id" -> region.sa4Id.toJson,
+            "sa3Id" -> region.sa3Id.toJson,
+            "sa2Id" -> region.sa2Id.toJson
+          ).filter(x => !x._2.equals(JsNull)))
+
+      private def convertOptionalStringField(
+          json: JsObject,
+          fieldName: String): Option[String] = {
+        json.getFields(fieldName).headOption.flatMap { fieldValue =>
+          fieldValue match {
+            case JsNull => None
+            case _      => Some(fieldValue.convertTo[String])
+          }
+        }
+      }
 
       override def read(jsonRaw: JsValue): Region = {
         val json = jsonRaw.asJsObject
 
         Region(
           QueryRegion(
             regionId = json.getFields("regionId").head.convertTo[String],
-            regionType = json.getFields("regionType").head.convertTo[String]),
-            regionName = json.getFields("regionName").headOption.map(_.convertTo[String]),
-            boundingBox = json.getFields("boundingBox").headOption.map(_.convertTo[BoundingBox](bbFormat)),
-            regionShortName = json.getFields("regionShortName").headOption.flatMap { shortName => shortName match {
-              case JsNull => None
-              case _ => Some(shortName.convertTo[String])
-            }})
+            regionType = json.getFields("regionType").head.convertTo[String]
+          ),
+          regionName =
+            json.getFields("regionName").headOption.map(_.convertTo[String]),
+          boundingBox = json
+            .getFields("boundingBox")
+            .headOption
+            .map(_.convertTo[BoundingBox](bbFormat)),
+          regionShortName = convertOptionalStringField(json, "regionShortName"),
+          steId = convertOptionalStringField(json, "steId"),
```

Comment:
Is there any way we can make this work without having to directly reference Australian region types in the code?

My concern is that this works OK within Australia, but if someone outside of Australia were to try to use Magda then not only would none of this work for them, but they'd suffer from this assumption that we're building into the code that all regions are australian ABS regions.

Reply:
Changed to the more generic names like `lv1Id` etc.